### PR TITLE
[5.5] Added str_compare and str_exactly helpers

### PR DESCRIPTION
**Pull request by azeemhassni**
### Two string helpers
1. **`str_exactly`** compares two strings.
2. **`str_compare`** compares two strings case insensitively.

### CODE REVIEW

File: src/Illuminate/Support/helpers.php [hunk 1]
```diff
@@ -874,6 +874,34 @@ function str_is($pattern, $value)
...
+     * @param $string
+     * @param  string $value
+     * @return bool
+     */
+    function str_exactly($string, $value)
+    {
+        return $string === $value;
+    }
+}
+
```
Comment (tillkruss) [inline src/Illuminate/Support/helpers.php]: This helper is not useful.

File: src/Illuminate/Support/helpers.php [hunk 2]
```diff
@@ -874,6 +874,34 @@ function str_is($pattern, $value)
...
+     * Determine if a given string is equal to an-other string ignoring the case.
+     *
+     * @param $string
+     * @param  string $value
+     * @return bool
+     */
+    function str_compare($string, $value)
+    {
+        return Str::lower($string) === Str::lower($value);
+    }
```
Comment (tillkruss) [inline src/Illuminate/Support/helpers.php]: You can use `strcasecmp()` here.

**Comment by GrahamCampbell** [comment]
Thanks, but I don't think we need to bloat the core with more functions. `str_exactly` is the same as `===`.

File: src/Illuminate/Support/helpers.php [hunk 3]
```diff
@@ -874,6 +874,34 @@ function str_is($pattern, $value)
     }
 }
 
+if (! function_exists('str_compare')) {
+    /**
+     * Determine if a given string is equal to an-other string ignoring the case.
+     *
+     * @param $string
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/helpers.php]: Invalid syntax.

File: src/Illuminate/Support/helpers.php [hunk 4]
```diff
@@ -874,6 +874,34 @@ function str_is($pattern, $value)
...
+    {
+        return strcasecmp($string, $value) === 0;
+    }
+}
+
+if (! function_exists('str_exactly')) {
+    /**
+     * Determine if a given string is equal to the second string.
+     *
+     * @param $string
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/helpers.php]: Invalid syntax.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by azeemhassni** [comment]
@GrahamCampbell  I understand but I think the code looks more clean as compare to the `===` and `strtolower('something') === strtolower('SomeThing')` statements.

**Comment by sisve** [comment]
These are implemented badly. You cannot compare strings in a case insensitive way without also working with locale information. You need locale rules to determine if two letters are the same. The uppercase version if "i" (Latin Small Letter I, U+0069) is "İ" (Latin Capital Letter I with Dot Above, U+0130) in Turkish, not "I" (Latin Capital Letter I, U+0049) as you are probably used to. Wikipedia has an article on this; [Dotted and dotless I](https://en.wikipedia.org/wiki/Dotted_and_dotless_I).

The documentation should at least state that the functionality is only guaranteed/tested for ASCII.

**Pull request closed without merging**